[Albert Sutiono] iP

albertsutz opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))


DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- FAST SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**


Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (Coming Soon!)

If you are a Java Programmer, you can use it to practice Java too. Here's the `main` method:
```java

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

albertsutz committed:
Update .gitignore to exclude Java executables

albertsutz committed:
Implemented basic Functionalities (greet, echo, exit) for Level 1

albertsutz committed:
Enables user add task and list them (Level 2)

albertsutz committed:
Implement class Task

albertsutz committed:
Integrated class Task with Level 2

albertsutz committed:
Add mark and unmark commands to Duke as Level 3

albertsutz committed:
Implement classes Event, Todo, Deadline as subclass of Task

albertsutz committed:
Integrated Todo, Deadline, and Event with Duke (Level 4)

albertsutz committed:
Added test cases for Level-4

albertsutz committed:
fixed small bugs

albertsutz committed:
Added Basic Exception for Invalid Command and Argument

albertsutz committed:
Integrated basic Exceptions with Duke

albertsutz committed:
Fixed todo Exception bugs

albertsutz committed:
Add Delete command (Level 6)

albertsutz committed:
Improve introduction UI

albertsutz committed:
Refactor Task and Implement Factory constructor for Event, Deadline, and Todo

albertsutz committed:
Refactor Duke.java

albertsutz committed:
Implement Exception for invalid index

albertsutz committed:
Ignore temporary data directory

albertsutz committed:
Implement factory constructor for Deadline, Event, and Todo using List

albertsutz committed:
Implement factory constructor for Task

albertsutz committed:
Implement initial save command (Level-7)

albertsutz committed:
Implement date parser in Deadline

albertsutz committed:
Merge branch 'branch-Level-7' into master

albertsutz committed:
Merge branch 'branch-Level-8' into master

# Conflicts:
#	src/main/java/Deadline.java

albertsutz committed:
Abstract constants into Constants class

albertsutz committed:
Implement Ui class to follow OOP standard

albertsutz committed:
Implement TasksList as main storage to follow OOP standards

albertsutz committed:
Implement Storage for importing and exporting tasks history (A-MoreOOP)

albertsutz committed:
Fix TasksList response to improve UI

albertsutz committed:
Remove unnecessary imports

albertsutz committed:
Refactor and organize code into packages (A-Packages)

albertsutz committed:
Update Constant class to include commands word list

albertsutz committed:
Implement Parser

albertsutz committed:
Add method to get the size of taskslist

albertsutz committed:
Add JUnit tests for Parser, TasksList, and Task

albertsutz committed:
Create Jar manifest for duke

albertsutz committed:
Adds JavaDoc for all public methods

albertsutz committed:
Fix coding style

albertsutz committed:
Implement find command

albertsutz committed:
Add JavaDoc to find method

vishandi commented on src/main/java/duke/TasksList.java:
```diff
@@ -151,6 +151,13 @@ public String importStorageStrings(List<String> tasksStrings) throws InvalidArgu
+     * @return Tasks in text format that contains keyword to be found.
+     * @throws InvalidArgumentException If there is no keyword inputted.
+     */
     public String find(List<String> instruction) throws InvalidArgumentException {
```
This is fine, but I think you could increase readability by changing the method name to "findMatchingTasks" or something along that line.

vishandi commented on src/main/java/duke/TasksList.java:
```diff
@@ -150,4 +150,23 @@ public String importStorageStrings(List<String> tasksStrings) throws InvalidArgu
+        StringBuilder response = new StringBuilder("");
+
+        for (int i = 0; i < tasks.size(); i++) {
+            if (tasks.get(i).toString().contains(instruction.get(1))) {
```
Do you mean `tasks.get(i).getName()...`?

vishandi commented on src/main/java/duke/TasksList.java:
```diff
@@ -0,0 +1,179 @@
+     *
+     * @return List of tasks with numberings in text format.
+     */
+    public String list() {
```
To increase readability, you could consider changing the method name to "getListOfTasks".

vishandi commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,39 @@
+    /**
+     * Prints initial greetings when the app starts.
+     */
+    public void initialGreet() {
```
As per Java coding standard, you could change this method name into something that starts with a verb, for example "greetUser".

vishandi commented on src/main/java/duke/exception/InvalidArgumentException.java:
```diff
@@ -0,0 +1,12 @@
+package duke.exception;
+
+public class InvalidArgumentException extends DukeException {
+
+    /**
+     * Constructs the InvalidArgumentException class.
+     * This exception occurs when the user input format is invalid but with some known parts.
+     */
+    public InvalidArgumentException() {
+        super("Hmm, I recognized your command but it seems like the arguments are not complete / invalid.");
+    }
```
This is fine and the abstraction is meaningful. One suggestion that I have is that you could make all the sub-exceptions as static functions in `DukeException.java`. This way, all the exceptions are stored in the same class so it is neater. But, at the end of the day, it is only the matter of preference :).

vishandi commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+     * @param date Input value to validate.
+     * @return True if the input value is a date, false otherwise.
+     */
+    private Boolean dateValidator(String date) {
```
Can consider changing the method name to `isValid(String date)`.

vishandi commented on src/main/java/duke/util/Constants.java:
```diff
@@ -0,0 +1,20 @@
+public class Constants {
+    public static final String LINE = "___________________________________________________________________________";
+    public static final String GAP = "      ";
+    public static final String LOGO =
```
Nice logo!

vishandi reviewed (commented):
Overall, I think your code is great! I could understand the reason why you structure your code as such. Though, some changes could be made (especially about the method names) that could improve readability.

albertsutz committed:
Add JavaDoc documentation for all public classes

albertsutz committed:
Fix find-related methods to improve readability

albertsutz committed:
Fix coding style to follow coding standard

albertsutz committed:
Merge branch 'branch-A-CodingStandard' into master

albertsutz committed:
Merge branch 'branch-A-JavaDoc' into master

albertsutz committed:
Merge branch 'branch-Level-9' into master

albertsutz committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support

albertsutz committed:
Integrate gradle to build and run Duke

albertsutz committed:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Integrate gradle to build and run Duke

albertsutz committed:
Configure CheckStyle using Gradle for automation

albertsutz committed:
Fix checkstyle issues

albertsutz committed:
Merge branch 'branch-A-CheckStyle'

* branch-A-CheckStyle:
  Fix checkstyle issues
  Configure CheckStyle using Gradle for automation

albertsutz committed:
Include JavaFX dependencies to gradle

albertsutz committed:
Add Resources for GUI

albertsutz committed:
Implement Basic GUI for Duke

albertsutz committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Implement Basic GUI for Duke
  Add Resources for GUI
  Include JavaFX dependencies to gradle

albertsutz committed:
Fix MainClassName issue

albertsutz committed:
Improve code quality : deep nesting in Parser

Class Parser has a code quality issue with regards to deep nesting.

Using if else statement in this case will confuse reader. There is also an issue of deep nesting within the if else statements which is a code quality violation.

Let's refactor the code and use switch case instead to represent the conditional statements. The implementation of switch case and if statements does not need much change after all.

Switch case will be more intuitive to other developers as it clearly states what the essence of the conditions.

albertsutz committed:
Improve code quality : deep nesting in Task

Class Task has a code quality issue with regards to deep nesting.

Using if else statement in this case will confuse reader. There is also an issue of deep nesting within the if else statements which is a code quality violation.

Let's refactor the code and use switch case instead to represent the conditional statements. The implementation of switch case and if statements does not need much change after all.

Switch case will be more intuitive to other developers as it clearly states what the essence of the conditions.

albertsutz committed:
Remove indentation for switch case statements to follow coding style convention

albertsutz committed:
Include assertions to make sure tasks command execution.

albertsutz committed:
Merge pull request #2 from albertsutz/branch-A-Assertions

Include assertions to make sure tasks command execution.

albertsutz committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Include assertions to make sure tasks command execution.

albertsutz committed:
Merge pull request #3 from albertsutz/branch-A-CodeQuality

Branch a code quality

albertsutz committed:
Implement reminder feature to let users list upcoming deadlines easily

albertsutz committed:
Merge branch 'branch-Reminders'

* branch-Reminders:
  Implement reminder feature to let users list upcoming deadlines easily

albertsutz committed:
Improve GUI : Include greetings and proper good bye, improve layout for better user experience

albertsutz committed:
Fix bugs : Ui still printing even when GUI launcher is used

albertsutz committed:
Implement saving feature after every command

albertsutz committed:
Change .jar file name

albertsutz committed:
Fix bugs and checkstyle issues

albertsutz committed:
Implement help command

albertsutz committed:
Write User Guide

albertsutz committed:
Fix bugs : dialog box being truncated

albertsutz committed:
Update User Guide

albertsutz committed:
Fix Bug: reminder numbering

albertsutz committed:
Set theme jekyll-theme-minimal

albertsutz committed:
Fix file name

albertsutz committed:
Set theme jekyll-theme-cayman

albertsutz committed:
Fix typo in README

albertsutz committed:
Fix typo

albertsutz committed:
Implement bye, list, and help command

albertsutz committed:
Implement add, find, mark, reminder, save, and unmark commands

albertsutz committed:
Refactor code: implement each command as Command Classes to preserve OOP

albertsutz committed:
Fix checkstyle

albertsutz committed:
Merge branch 'refactor-commands'

* refactor-commands:
  Fix checkstyle
  Refactor code: implement each command as Command Classes to preserve OOP
  Implement add, find, mark, reminder, save, and unmark commands
  Implement bye, list, and help command

albertsutz committed:
Refactor code: Parser, Storage, TasksList, and Ui should be in Duke's utility package

albertsutz committed:
Fix checkstyle: wrong import order

albertsutz committed:
Give small comments and clean up spaces

albertsutz committed:
Refactor code: change exportData return type to void

albertsutz committed:
Clean up code

albertsutz committed:
Merge branch 'refactor-commands'

* refactor-commands:
  Clean up code
  Refactor code: change exportData return type to void
  Give small comments and clean up spaces
  Fix checkstyle: wrong import order
  Refactor code: Parser, Storage, TasksList, and Ui should be in Duke's utility package
